brainframe_qt/ui/brainframe_app.py

```diff
@@ -177,52 +177,50 @@ def _shutdown(self):
         self.connection_manager.terminate()
 
     def _start_ui(self):
-        self.verify_version_match()
+        # Start the client if not already started
+        if not self.main_window:
 
-        ExtensionLoader().load_extensions()
+            def verify_version_before_starting(version: str):
+                if version != brainframe_qt.__version__:
+                    self._handle_version_mismatch(version)
 
-        self.main_window = MainWindow()
-        self.main_window.show()
+                ExtensionLoader().load_extensions()
 
-        self.splash_screen.finish(self.main_window)
+                self.main_window = MainWindow()
+                self.main_window.show()
+
+                self.splash_screen.finish(self.main_window)
+
+            worker = QTAsyncWorker(self, api.version,
```

Comment:
There's no way that two threads can run _start_ui at once, right? If so, then there would be a race condition where two threads could get past the `if not self.main_window` line

I'm assuming _start_ui is called from the QT eventloop, so only a single thread can do it.

Reply:
Actually, I think there might be a race condition. 

Thread 1: 
- Gets past `if not self.main_window`
- Starts a QTAsyncWorker

Thread 2: 
- Get's past `if not self.main_window`
- Starts a QTAsyncWorker

Is there a benefit of doing the `verify_version_before_starting` code in a QTAsyncWorker in the first place?

Reply:
Almost missed this comment. I guess there's not really a problem with doing the version check synchronously because the UI hasn't started yet. But I would like it to eventually be asynchronous